### PR TITLE
Made tableam.h file dependency free

### PR DESCRIPTION
**Pull request by yxu2162**
Made tableam.h dependency free. This technically makes the include/columnar directory dependency free but there are still some header files that don't have any citus dependencies but are in the distributed directory (pg_version_constants.h and listutils.h)

### CODE REVIEW

File: src/include/columnar/columnar_tableam.h [hunk 1]
```diff
@@ -51,6 +49,9 @@ typedef struct ColumnarScanDescData *ColumnarScanDesc;
 const TableAmRoutine * GetColumnarTableAmRoutine(void);
 extern void columnar_tableam_init(void);
 extern void columnar_tableam_finish(void);
+extern void CreateTruncateTrigger(Oid relationId);
+extern void EnsureTableOwner(Oid relationId);
+extern bool CheckCitusVersion(int elevel);
```
Comment (onurctirtir) [inline src/include/columnar/columnar_tableam.h:54]: got confused a bit, where do we define those under columnar/ directory ?

File: src/include/columnar/columnar_tableam.h [hunk 2]
```diff
@@ -51,6 +49,9 @@ typedef struct ColumnarScanDescData *ColumnarScanDesc;
 const TableAmRoutine * GetColumnarTableAmRoutine(void);
 extern void columnar_tableam_init(void);
 extern void columnar_tableam_finish(void);
+extern void CreateTruncateTrigger(Oid relationId);
```
Comment (jeff-davis) [inline src/include/columnar/columnar_tableam.h:52]: Looks like this is not referenced or defined in `columnar/`, so I think this line can be removed.

File: src/include/columnar/columnar_tableam.h (hunk 1 above)
Comment (jeff-davis) [inline src/include/columnar/columnar_tableam.h:54]: These two are defined in the `distributed/` directory, and referenced in the `citus/` directory, so they seem to be dependencies that we need to fix.

**Comment by yxu2162** [comment]
I made CheckCitusVersion a hook in columnar since we only needed it when Columnar was working with Citus. I referenced https://github.com/citusdata/citus/pull/5547/files as a guide on how to make a hook in Citus. Was unsure if I needed to move CheckCitusVersion() into utility_hook like what Jeff did.

File: src/backend/columnar/columnar_tableam.c [hunk 3]
```diff
@@ -2454,7 +2515,11 @@ alter_columnar_table_reset(PG_FUNCTION_ARGS)
 							   quote_identifier(RelationGetRelationName(rel)))));
 	}
 
-	EnsureTableOwner(relationId);
+	if (!pg_class_ownercheck(relationId, GetUserId()))
+	{
+		aclcheck_error(ACLCHECK_NOT_OWNER, OBJECT_TABLE,
+					   get_rel_name(relationId));
```
Comment (jeff-davis) [inline src/backend/columnar/columnar_tableam.c:2521]: You can add a SQL test so that this code is covered.

Comment (yxu2162) [reply src/backend/columnar/columnar_tableam.c:2521]: Added in multi_multiuser.sql

File: src/backend/columnar/columnar_tableam.c [hunk 4]
```diff
@@ -1024,13 +1051,16 @@ static void
...
-		 */
-		return;
+		if (!ColumnarCheckCitusVersion_hook(WARNING))
+		{
+			/*
+			 * Skip if the extension catalogs are not up-to-date, but avoid
+			 * erroring during auto-vacuum.
+			 */
+			return;
+		}
```
Comment (thanodnl) [inline src/backend/columnar/columnar_tableam.c:1063]: This comment was quite enlightening.

I guess a lot of the `CheckCitusVersion` invocations were to guard against citus specific functions being called without insurance that the extension had been created and/or upgraded correctly.

How will columnar be handling these cases?

I am asking this to understand if we actually need this hook here, or that columnar will have other guards to make sure the catalogs exist in the correct state.

Given the catalog tables for columnar and citus are sufficiently separated I wonder if we can simple remove the invocations here, or that we need comparable functionality in columnar which will superseed the citus version checks from these call sites.

**Pull request closed without merging**